### PR TITLE
fix: [And-Or] fixed And and Or operation for const

### PR DESCRIPTION
**Pull request by jiekun**
### Describe Your Changes

Fix https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6637

Currently, the `And` operation (e.g. `1 and NaN`) always returns the `left` expression. For expressions that do not contain an intersection, it should return an **empty** result. The same principle applies to the `Or` operation.
- `And` return `left` if `left` and `right` is not `NaN`. Otherwise, `NaN` is returned.
- `Or` return the first non-`NaN` item. If both `left` and `right` are `NaN`, it returns `NaN`.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: binaryop/funcs.go [hunk 1]
```diff
@@ -107,3 +107,21 @@ func Ifnot(left, right float64) float64 {
...
+		return nan
+	}
+	return left
+}
+
+// Or return the first non-NaN item. If both left and right are NaN, it returns NaN.
+func Or(left, right float64) float64 {
+	if !math.IsNaN(left) {
+		return left
+	} else if !math.IsNaN(right) {
```
Comment (valyala) [inline binaryop/funcs.go:123]: It is OK to just return `right` here without checking it for `nan`.

File: binaryop/funcs.go [hunk 2]
```diff
@@ -107,3 +107,21 @@ func Ifnot(left, right float64) float64 {
 	}
 	return nan
 }
+
+// And return left if left and right is not NaN. Otherwise, NaN is returned.
+func And(left, right float64) float64 {
```
Comment (valyala) [inline binaryop/funcs.go:112]: Could you verify other places in metricsql package and in the VictoriaMetrics package, which use the `binaryop.*` functions? Probably, they also need to be updated with the newly added `And` and `Or` functions?

**Review by valyala — APPROVED** [review]
Looks good overall. See the two comments.

File: binaryop/funcs.go (hunk 1 above)
Comment (valyala) [reply binaryop/funcs.go:123]: Addressed this in the follow-up commit - 363400dae5e18e099a6b8bb87dbedbb0a3be08c6

**Comment by jiekun** [comment]
Thanks!

**Pull request merged**